modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/executor/MethodArgumentResolverDelegate.java

```diff
@@ -96,7 +98,7 @@ else if (MuleMessage.class.isAssignableFrom(parameterType))
             {
                 argumentResolver = MESSAGE_ARGUMENT_RESOLVER;
             }
-            else if (annotations.containsKey(ParameterGroup.class) || annotations.containsKey(MetadataKeyId.class))
+            else if (isParameterFieldContainer(annotations.keySet(), typeLoader.load(parameterType)))
```

Comment:
this method name doesn't really reflect what the method does. What's a parameter field container?